### PR TITLE
Fix `costEx/Sz/level` for `GT_STORE_DYN_BLK`.

### PR DESCRIPTION
**Pull request by sandreenko**
Another issue found by PVS, we were reassigning `costEx, costSz` without using `GT_STORE_DYN_BLK` values.

There were no diffs from that change, so I tried to find how can we hit this code path. It looks like we create `GT_STORE_DYN_BLK`  only here:
https://github.com/dotnet/coreclr/blob/79319d50d45a95375f26f60341575e0284c3fa00/src/jit/rationalize.cpp#L462-L464 and that is LIR so it after possible calls to `gtSetEvalOrder`.
Is it correct or have I missed something, @CarolEidt @mikedn ?
If it is correct that probably we need to delete 
`case GT_STORE_DYN_BLK:` from `gtSetEvalOrder` so it goes to `default: Unexpected Op`.

Link #7027.

### CODE REVIEW

**Comment by CarolEidt** [comment]
It is entirely unlikely that we will hit this path, however I'm not certain that we should assert. We can call `gtSetEvalOrder` from `Lowering` in `LIR::SeqTree`. What makes it extremely unlikely is that I can't think of a situation where `Lowering` would create a new block store with a non-constant block size. That said, it wouldn't be an invalid thing to do theoretically speaking.

**Comment by CarolEidt** [comment]
The other thing to note is that `GT_DYN_BLK` itself is pretty rare.

File: src/jit/gentree.cpp [hunk 1]
```diff
@@ -4484,18 +4484,19 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
         {
             costEx = 0;
```
Comment (mikedn) [inline src/jit/gentree.cpp]: These should be deleted

**Comment by mikedn** [comment]
> It is entirely unlikely that we will hit this path,

Not only that but the consequences of hitting this path and getting it wrong are likely to be almost invisible. The costs are mostly a CSE/loop hoisting thing so if you're building new IR in lowering and get the costs wrong nothing happens. In fact they're already wrong for a bunch of such IR because `SeqTree` is not always used (and IMO it should be avoided whenever possible).

And both costs and level are approximations (and not very good ones) that should only affect CQ, not correctness.

Anyway, `gtCanSwapOrder` is one of those things inherited from JIT32 that do not make a lot of sense in RyuJIT. I hope to dismantle it once I get rid of `GT_LIST`.

**Pull request merged**